### PR TITLE
Musark 972 - ViewObject

### PR DESCRIPTION
**Pull request by steinaol**
Missing a lot of fields, but this works with dataflow. New tasks will be created when the design is finished and committed by the reference group

### CODE REVIEW

File: src/components/events/eventTableComponent.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        columns={[
+          { key: 'eventDate', label: 'Dato' },
+          { key: 'eventType', label: 'Type hendelse' },
+          { key: 'registeredBy', label: 'Utført av' },
+          { key: 'keyData', label: 'Nøkkeldata' },
+          { key: 'note', label: 'Kommentar' },
+          { key: 'id', label: '' }
+        ]}
+        sortable={['eventType', 'eventDate', 'registeredBy', 'note' ]}
+        noDataText="Ingen hendelser funnet på objektet"
```
Comment (ingarabr) [inline src/components/events/eventTableComponent.jsx]: Should this text have been moved to the text file?

File: src/components/events/eventTableComponent.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      >
+        {events && events.map((event, i) =>
+          <Tr key={i}>
+            <Td column="eventDate">{event.eventDate}</Td>
+            <Td column="type">{event.eventType}</Td>
+            <Td column="registeredBy">{event.registeredBy}</Td>
+            <Td column="keyData">{event.keyData}</Td>
+            <Td column="note"><span>{event.note}</span></Td>
+            <Td column="id">Les mer</Td>
```
Comment (ingarabr) [inline src/components/events/eventTableComponent.jsx:32]: I guess this should be converted to a link in a future PR.

File: src/modules/objects/objectStore.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import { createStore, createAction } from 'react-rxjs/dist/RxStore';
```
Comment (ingarabr) [inline src/modules/objects/objectStore.js:2]: Should we add flow to this also?

File: src/modules/objectsearch/ObjectSearchComponent.js [hunk 4]
```diff
@@ -126,6 +126,7 @@ export class ObjectSearchComponent extends React.Component {
                               />
                             </a>
                           </th>
+                          <th>{Config.isDev && 'Se objekt'}</th>
```
Comment (ingarabr) [inline src/modules/objectsearch/ObjectSearchComponent.js:129]: I like the pattern but should we take it one step future.
If we add a object in the config where we can add feature toggles. Then we can easily enable/disable them. When we're happy with the feature it is easy to find all the places we need to remove the toggle check. 

You don't need to do this in this PR but I think we should discuss the pattern and decide if we want to use it or not.

File: src/modules/objectsearch/ObjectSearchComponent.js [hunk 5]
```diff
@@ -197,6 +198,21 @@ export class ObjectSearchComponent extends React.Component {
                                         />}
                                   </a>}
                               </td>
+                              <td>
+                                <a
```
Comment (ingarabr) [inline src/modules/objectsearch/ObjectSearchComponent.js]: This is a misuse of the `a` element. We should use the `button` element and style it as a link.

File: src/types/events.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+/* @flow */
+
+type T_EventData = {
```
Comment (ingarabr) [inline src/types/events.js]: Don't know we we concluded on the naming of the types. Anyway we should follow the same pattern.

I don't like to mark it as a type (prefix with `T_`). Little bit inspired by Martin (https://www.martinfowler.com/bliki/InterfaceImplementationPair.html)

**Review by ingarabr — CHANGES_REQUESTED** [review]

File: src/modules/objects/objectStore.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+          callBack: val.callBack
+        };
+
+        return Observable.forkJoin(
+          Observable.of(response),
+          Event.getAnalysesAndMoves()(v),
+          Sample.loadSampleDataForObject()(v)
+        ).map(([o, e, s]) => ({ objectData: o, events: e, samples: s }));
+      })
+      .flatMap(r => r);
```
Comment (ingarabr) [inline src/modules/objects/objectStore.js]: You can remove this if you change the first from `map` to `flatMap`

File: src/modules/objects/objectStore.js [hunk 8]
```diff
@@ -0,0 +1,41 @@
+import { createStore, createAction } from 'react-rxjs/dist/RxStore';
+import { Observable } from 'rxjs';
+import MusitObject from '../../models/object';
+import Sample from '../../models/sample';
+import Event from '../../models/event';
+export const initialState = { objectData: {}, events: [], samples: [] };
+
+const loadObjectData = () =>
+  val => {
+    return MusitObject.getObjectDetails()(val)
```
Comment (ingarabr) [inline src/modules/objects/objectStore.js]: I find it easier to read when we drop the return and change {} to (). `val => ( MusitObject.getObjectDetails()(val) ... )`

File: src/components/events/eventTableComponent.jsx (hunk 1 above)
Comment (steinaol) [reply src/components/events/eventTableComponent.jsx]: Yes, I agree. I talked with Line about it, and she agree'd that we should do it right now as we develop. The sooner the better. So I'll do it

File: src/components/events/eventTableComponent.jsx (hunk 2 above)
Comment (steinaol) [reply src/components/events/eventTableComponent.jsx:32]: Yep!

File: src/modules/objects/objectStore.js (hunk 3 above)
Comment (steinaol) [reply src/modules/objects/objectStore.js:2]: Yes, I'll try

File: src/modules/objectsearch/ObjectSearchComponent.js (hunk 5 above)
Comment (steinaol) [reply src/modules/objectsearch/ObjectSearchComponent.js]: Ok...:-)

File: src/types/events.js (hunk 6 above)
Comment (steinaol) [reply src/types/events.js]: I agree, we talked about this, I remember

File: src/modules/objects/objectStore.js (hunk 7 above)
Comment (steinaol) [reply src/modules/objects/objectStore.js]: I'll do it...

File: src/modules/objects/objectStore.js (hunk 8 above)
Comment (steinaol) [reply src/modules/objects/objectStore.js]: Ok!

File: src/modules/objects/objectStore.js (hunk 7 above)
Comment (steinaol) [reply src/modules/objects/objectStore.js]: I tried this, but it did not seem to work. We'll have to look closer at this tomorrow.

File: src/modules/objectsearch/ObjectSearchComponent.js (hunk 5 above)
Comment (steinaol) [reply src/modules/objectsearch/ObjectSearchComponent.js]: I will create a new task for this
[](https://jira.usit.uio.no/browse/MUSARK-995)

Comment (steinaol) [reply src/modules/objectsearch/ObjectSearchComponent.js]: [https://jira.usit.uio.no/browse/MUSARK-995](url)

**Review by ingarabr — APPROVED** [review]

**Pull request merged**